File: CPP/Komputer.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"
#include "Ssd.cpp"


using namespace std;


class Komputer
{
private:
    string nama;
    Cpu cpu;
    vector<Ram> ramList;
    Harddrive harddrive;
    vector<Ssd> ssdList;
public:
    Komputer()
    {


    }


    Komputer(string nama, Cpu cpu, vector<Ram> ramList, Harddrive harddrive, vector<Ssd> ssdList)
    {
        this->nama = nama;
        this->cpu = cpu;
        this->ramList = ramList;
        this->harddrive = harddrive;
        this->ssdList = ssdList;
    }


    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setCpu(Cpu cpu)
    {
        this->cpu = cpu;
    }


    void setRam(vector<Ram> ramList)
    {
        this->ramList = ramList;
    }    
   
    void setHarddrive(Harddrive harddrive)
    {
        this->harddrive = harddrive;
    }


    void addRam(Ram ram)
    {
        this->ramList.push_back(ram);
    }

    void addSsd(Ssd ssd)
    {
        this->ssdList.push_back(ssd);
    }


    string getNama()
    {
        return this->nama;
    }


    Cpu getCpu()
    {
        return this->cpu;
    }


    vector<Ram> getRamList() // return vector
    {
        return this->ramList;
    }


    Harddrive getHarddrive()
    {
        return this->harddrive;
    }

    vector<Ssd> getSsdList()
    {
        return this->ssdList;
    }


    ~Komputer()
    {


    }
};

File: Python/Komputer.py
from Cpu import Cpu
from Harddrive import Harddrive

class Komputer:
    def __init__(self, nama="", cpu=None, ram_list=None, harddrive=None, ssd_list=None):
        self.nama = nama
        self.cpu = cpu if cpu else Cpu()
        self.ram_list = ram_list if ram_list else []
        self.harddrive = harddrive if harddrive else Harddrive()
        self.ssd_list = ssd_list if ssd_list else []
    
    def set_nama(self, nama):
        self.nama = nama
    
    def set_cpu(self, cpu):
        self.cpu = cpu
    
    def set_ram(self, ram_list):
        self.ram_list = ram_list
    
    def set_harddrive(self, harddrive):
        self.harddrive = harddrive
    
    def add_ram(self, ram):
        self.ram_list.append(ram)
    
    def add_ssd(self, ssd):
        self.ssd_list.append(ssd)
    
    def get_nama(self):
        return self.nama
    
    def get_cpu(self):
        return self.cpu
    
    def get_ram_list(self):
        return self.ram_list
    
    def get_harddrive(self):
        return self.harddrive
    
    def get_ssd_list(self):
        return self.ssd_list
